packages/connectivity/src/scp-cf/resilience/resilience.ts

```diff
@@ -0,0 +1,233 @@
+    exitChain: middlewareInOutOptions.exitChain
+  });
+}
+
+/**
+ * TODO: Add JSDoc later.
+ * @param resilienceMiddlewareOptions - TODO: Add JSDoc later.
+ * @returns TODO: Add JSDoc later.
+ * @internal
+ */
+export function createCircuitBreakerMiddleware<T>(
+  resilienceMiddlewareOptions: ResilienceMiddlewareOptions
+): Middleware<T> {
+  const { circuitBreaker } = resilienceMiddlewareOptions;
+  const timeoutMiddlewareFn = (
+    middlewareInOutOptions: MiddlewareInOutOptions<T>
+  ) => {
+    const circuitBreakerOptions = circuitBreaker();
+
+    if (circuitBreakerOptions) {
+      return createCircuitBreaker({
+        ...circuitBreakerOptions
+      }).fire(middlewareInOutOptions.fn);
+    }
+
+    return middlewareInOutOptions.fn();
+  };
+
+  return (middlewareInOutOptions: MiddlewareInOutOptions<T>) => ({
+    fn: () => timeoutMiddlewareFn(middlewareInOutOptions),
+    exitChain: middlewareInOutOptions.exitChain
+  });
+}
+
+/**
+ * TODO: Add JSDoc later.
+ * @param resilienceMiddlewareOptions - TODO: Add JSDoc later.
+ * @returns TODO: Add JSDoc later.
+ * @internal
+ */
+export function createResilienceMiddlewareList<T>(
+  resilienceMiddlewareOptions: ResilienceMiddlewareOptions
+): Middleware<T>[] {
+  // From innermost to outtermost middleware:
+  // e.g., [a, b, c] -> c(b(a))
+  // The order of Retry(CB(Timeout)) should be [Timeout, CB, Retry], which makes sense.
+  return [
+    createTimeoutMiddleware(resilienceMiddlewareOptions),
+    createCircuitBreakerMiddleware(resilienceMiddlewareOptions)
+  ];
+}
+
+/**
+ * TODO: Add JSDoc later.
+ * @param resilienceMiddlewareOptions - TODO: Add JSDoc later.
+ * @returns TODO: Add JSDoc later.
+ * @internal
+ */
+export function createResilienceMiddleware<T>(
+  resilienceMiddlewareOptions: ResilienceMiddlewareOptions
+): Middleware<T> {
+  const resilienceMiddlewares: Middleware<T>[] = createResilienceMiddlewareList(
+    resilienceMiddlewareOptions
+  );
+  const reducedMiddleware = resilienceMiddlewares.reduce(
+    (prev, curr) => (middlewareInOutOptions: MiddlewareInOutOptions<T>) => {
+      if (middlewareInOutOptions.exitChain) {
+        return prev(middlewareInOutOptions);
+      }
+      return curr(prev(middlewareInOutOptions));
+    }
+  );
+  resilienceMiddlewareMap.set(resilienceMiddlewareOptions.id, {
+    middleware: reducedMiddleware,
+    options: resilienceMiddlewareOptions
+  });
+
+  return reducedMiddleware;
+}
+
+/**
+ * TODO: Add JSDoc later.
+ * @param id - TODO: Add JSDoc later.
+ * @returns TODO: Add JSDoc later.
+ */
+export function getResilienceMiddleware(
+  id: string
+): Middleware<any> | undefined {
+  return resilienceMiddlewareMap.get(id)?.middleware;
+}
+
+/**
+ * TODO: Add JSDoc later.
+ * @param resilienceMiddlewareOptions - TODO: Add JSDoc later.
+ * @returns TODO: Add JSDoc later.
+ */
+export function resilience(
+  resilienceMiddlewareOptions: ResilienceMiddlewareOptions
+): Middleware<any> {
```

Comment:
[req] Add a generic type instead of <any>

Reply:
Done.